src/main/java/org/opensearch/securityanalytics/rules/objects/SigmaDetectionItem.java

```
@@ -111,7 +112,14 @@ public static <T> SigmaDetectionItem fromMapping(String key, Either<T, List<T>>
 
         List<SigmaType> sigmaTypes = new ArrayList<>();
         for (T v: values) {
-            sigmaTypes.add(SigmaTypeFacade.sigmaType(v));
+            SigmaType sigmaType = SigmaTypeFacade.sigmaType(v);
+            // throws an error if sigmaType is an empty string and the modifier is "contains" or "startswith" or "endswith"
+            boolean invalidModifierWithEmptyString = modifierIds.contains("contains") || modifierIds.contains("startswith") || modifierIds.contains("endswith");
+            if (sigmaType.getClass().equals(SigmaString.class) && v.toString().isEmpty() && invalidModifierWithEmptyString) {
```

What will happen to the other values? Should we skip the empty strings and use values which are present in the list?

---

We decided to throw an exception instead of skipping the empty string here because if we allowed the rule creation to go through, the empty string would still appear in the rule itself but could not be turned into a meaningful/correct query. 

Instead we added additional parsing logic on the front end to ensure that empty strings get parsed out when creating a rule from the UI. This PR is an additional check if the rule gets created from the API directly.